src/ssort/_requirements.py

```diff
@@ -52,12 +53,22 @@ def _get_requirements_for_function_def(
     for decorator in node.decorator_list:
         yield from get_requirements(decorator)
 
-    yield from get_requirements(node.args)
+    scope: set[str] = set()
+    if sys.version_info >= (3, 12):
+        for type_param in node.type_params:
+            yield from get_requirements(type_param)
+        scope.update(type_param.name for type_param in node.type_params)  # type: ignore[attr-defined]
```

Comment:
I've added the `# type: ignore[attr-defined]` here an in a couple of lines below because mypy complained that `"type_param" has no attribute "name"`. But type_param is `TypeVar | ParamSpec | TypeVarTuple` and all the latter have a name.